test: BDD tests for DIDExchange through DIDComm Router/Mediator

rolsonquadras opened the pull request:
- BDD test for flow described in https://wiki.hyperledger.org/display/ARIES/DIDComm+MediatorRouter

closes #950 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

rolsonquadras commented on pkg/didcomm/common/model/envelope.go:
```diff
@@ -0,0 +1,15 @@
+package model
+
+// Envelope for the DIDComm transport messages.
+type Envelope struct {
```
This is ideally catch-all for different envelop data models to be used in [Forward message](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0094-cross-domain-messaging/README.md#corerouting10forward) `msg` field.

sudeshrshetty commented on test/bdd/features/aries_router_e2e_sdk.feature:
```diff
@@ -0,0 +1,65 @@
+Feature: DIDComm Transport between two Edge Agents(without Inbound) through DIDComm Routers
+
+  # https://wiki.hyperledger.org/display/ARIES/DIDComm+MediatorRouter
+  Scenario: Decentralized Identifier(DID) Exchange between two Edge Agents through Routers
```
Minor : Should use `When, then ...` instead of just `And, And.....`

rolsonquadras replied:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/638de37c7ee7563d11e6d71efccf8bbba444de00..e75dea7c58e3b3d0ea308f21a689395d82016bd3#diff-6c628f852b19b92aa1ffe149b1485119R20).

rolsonquadras committed:
test: BDD tests for DIDExchange through DIDComm Router/Mediator

- BDD test for flow described in https://wiki.hyperledger.org/display/ARIES/DIDComm+MediatorRouter

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>
